include/stdlib.h

```diff
@@ -12,105 +12,785 @@ extern "C" {
+ * (unless another allocation function happened to result in a pointer value equal to ptr)
+ *
+ * @param ptr pointer to the memory to deallocate
+ * @return none
+ * */
+void free(void *ptr);
 
-void* calloc(size_t num, size_t size);
+/**
+ * @brief Allocates memory for an array of given number objects of size
+ *        and initializes all bytes in the allocated storage to zero.
+ *
+ * Allocates memory for an array of num objects of size size and initializes
+ * all bytes in the allocated storage to zero.
+ *
+ * @param num number of objects
+ * @param size size of the array
+ * @return (None)
+ * */
+void *calloc(size_t num, size_t size);
 
 /**
-* realloc(ptr, 0) is a special case that can be handled in two ways. See:
-*	 http://pubs.opengroup.org/onlinepubs/9699919799/functions/realloc.html
-* realloc(ptr, 0) returns NULL.
-*/
-void* realloc(void* ptr, size_t size);
+ * @brief Reallocates the given area of memory.
+ *
+ * Reallocates the given area of memory. It must be previously allocated by @see malloc,
+ * @see calloc or @see realloc and not yet freed with a call to @see free or @see realloc.
+ * Otherwise, the results are undefined.
+ *
+ * The reallocation is done by either:
+ *          1) expanding or contracting the existing area pointed to by ptr, if possible.
+ *             The contents of the area remain unchanged up to the lesser of the new and old sizes.
+ *             If the area is expanded, the contents of the new part of the array are undefined.
+ *          2) allocating a new memory block of size size bytes, copying memory area with size
+ *             equal the lesser of the new and the old sizes, and freeing the old block.
+ *
+ * If there is not enough memory, the old memory block is not freed and null pointer is returned.
+ * If ptr is NULL, the behavior is the same as calling malloc(size).
+ *
+ * If size is zero, the behavior is implementation defined (null pointer may be returned
+ * (in which case the old memory block may or may not be freed), or some non-null pointer may
+ * be returned that may not be used to access storage).
+ *
+ * @param ptr pointer to the memory area to be reallocated
+ * @param size new size of the array
+ * @return none
+ * */
+void *realloc(void *ptr, size_t size);
 
 /**
-* reallocf is a FreeBSD extension to realloc that frees
-* the input pointer if an error occurrs
-*
-* I do not hand the BSD case of `realloc(ptr,0)` freeing the ptr
-*/
-void* reallocf(void* ptr, size_t size);
+ * @brief Reallocates the given area of memory.
+ *
+ * Reallocates the given area of memory. It must be previously allocated by @see malloc,
+ * @see calloc or @see realloc and not yet freed with a call to @see free or @see realloc.
+ * Otherwise, the results are undefined.
+ *
+ * reallocf is a FreeBSD extension to realloc that frees
+ * the input pointer if an error occurrs
+ *
+ * I do not hand the BSD case of `realloc(ptr,0)` freeing the ptr
```

Comment:
"This library does not handle the BSD case where `realloc(ptr,0)` frees the ptr.